roles/grafana_cloud_operator/tasks/grafana_slo_hub_spoke.yml

```diff
@@ -0,0 +1,159 @@
+- name: Update CR status to start of slo dashboard for hubAndSpoke mode
+  operator_sdk.util.k8s_status:
+    api_version: grafanacloud.stakater.com/v1alpha1
+    kind: Config
+    name: "{{ cr_name }}"
+    namespace: "{{ cr_namespace }}"
+    status:
+      lastUpdated: "{{ ansible_date_time.iso8601 }}"
+      phase: "Initiated"
+      reason: "OperationStarted"
+      message: "Starting creation of SLO dashboard in hubAndSpoke mode"
+
+- name: Get all folders in Grafana Cloud
+  ansible.builtin.uri:
+    url: "{{ grafana_cloud_operator_slo_cloud_api }}/folders"
+    method: GET
+    body_format: json
+    status_code: [200, 202]
+    headers:
+      Authorization: Bearer {{ slo_api_token }}
+      Content-Type: "application/json"
+    return_content: true
+  register: grafana_cloud_folders
+
+- name: Debug grafana_cloud_folders
+  ansible.builtin.debug:
+    msg: "{{ grafana_cloud_folders }}"
+
+- name: Define existing folders for managed clusters
+  ansible.builtin.set_fact:
+    existing_folders: "{{ grafana_cloud_folders.json | selectattr('title', 'in', current_managed_clusters) | map(attribute='title') | list }}"
+
+- name: Define missing dashboards
+  ansible.builtin.set_fact:
+    missing_dashboards: "{{ current_managed_clusters | difference(existing_folders) }}"
+
+
+- name: Set the fact if the folder exists
+  ansible.builtin.set_fact:
+    folder_uid: "{{ item.uid }}"
+  when: item.title == missing_dashboards
+  with_items:
+    - "{{ grafana_cloud_folders.json }}"
+
+- name: Create Folder in Grafana Cloud
+  ansible.builtin.uri:
+    url: "{{ grafana_cloud_operator_slo_cloud_api }}/folders"
+    method: POST
+    return_content: true
+    body:
+      title: "{{ item }}"
+    body_format: json
+    headers:
+      Authorization: "Bearer {{ slo_api_token }}"
+      Content-Type: "application/json"
+    status_code: [200, 201]
+  register: grafana_cloud_folder
+  when: folder_uid is not defined
+  loop: "{{ missing_dashboards }}"
+  loop_control:
+    loop_var: item
+
+- name: Store the folder UID for each managed cluster
+  ansible.builtin.set_fact:
+    folder_uids: "{{ folder_uids | default({}) | combine({item: (grafana_cloud_folder.results | selectattr('item', 'equalto', item) | map(attribute='json') | first | default({})).uid }) }}" # yamllint disable-line rule:line-length
+  loop: "{{ missing_dashboards }}"
+  loop_control:
+    loop_var: item
+
+- name: Download SLO Dashboard YAML
+  ansible.builtin.get_url:
+    url: "https://raw.githubusercontent.com/stakater/charts/slo-observability-{{ gco_cr.spec.sloObservabilityVersion }}/slo-observability/templates/grafana/dashboards/slo-dashboard-cm.yaml" # yamllint disable-line rule:line-length
```

Comment:
1. So this is going to be a constant? if so, please put this in defaults.yml and fetch it here. 

2. If user wants to use their own dashboard, its not possible at this point? Does that require many changes in the actual creation process?

Reply:
1. not constant we our Config will have an option to change the version supplied here.
2. Moving it to main would require changes, since with current integration it won't be able to pick up {{ gco_cr.spec.sloObservabilityVersion }}

Reply:
Or we can have the entire URL in the config